app/src/main/java/com/apator/map/fragments/MapFragment.kt

```diff
@@ -0,0 +1,294 @@
+import org.koin.android.viewmodel.ext.android.viewModel
+import java.text.SimpleDateFormat
+import java.util.*
+import kotlin.collections.ArrayList
+
+class MapFragment : Fragment(), OnMapReadyCallback, PermissionsListener {
+    private val solarViewModel: SolarViewModel by viewModel()
+    private lateinit var mapView: MapView
+    private var isFabOpen = false
+    private lateinit var geoJson: GeoJsonSource
+    val bundle = Bundle()
+    private lateinit var mapboxMap: MapboxMap
+    private var permissionsManager: PermissionsManager = PermissionsManager(this)
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        val view = inflater.inflate(R.layout.fragment_map, container, false)
+        geoJson = GeoJsonSource("SOURCE_ID")
+        Mapbox.getInstance(context!!, R.string.API_KEY_MAPBOX.toString())
+
+
+
+        mapView = view.mapView
+        mapView.apply {
+            getMapAsync {
+                onMapReady(it)
+            }
+            onCreate(savedInstanceState)
+        }
+
+        val fab = view.findViewById<FloatingActionButton>(R.id.fab_more)
+        val fabsync = view.findViewById<FloatingActionButton>(R.id.fab_sync)
+        val fabreset = view.findViewById<FloatingActionButton>(R.id.fab_reset)
+        val fabsettings = view.findViewById<FloatingActionButton>(R.id.fab_settings)
+        fab.setOnClickListener {
+            when (isFabOpen) {
+                false -> showFabMenu(fab, fabsync, fabsettings, fabreset)
+                true -> hideFabMenu(fab, fabsync, fabsettings, fabreset)
+            }
+        }
+        fabsync.setOnClickListener {
+            Toast.makeText(context, "Synchronized", Toast.LENGTH_SHORT).show()
+            val current = Date()
+            val formatter = SimpleDateFormat("MMM dd yyyy HH:mma")
+            view.findViewById<TextView>(R.id.map_sync_date).text = formatter.format(current)
+
+            solarSync()
+        }
+        fabreset.setOnClickListener {
+            isFabOpen = false
+
+            if (mapboxMap.locationComponent.lastKnownLocation != null) targetCameraOnLocation()
+
+        }
+        fabsettings.setOnClickListener {
+            isFabOpen = false
+            view.findNavController().navigate(R.id.action_mapFragment_to_settingsFragment2)
+        }
+
+        return view
+    }
+
+    private fun targetCameraOnLocation() {
+        mapboxMap.animateCamera(
+            CameraUpdateFactory.newLatLngZoom(
+                LatLng(
+                    mapboxMap.locationComponent.lastKnownLocation!!.latitude,
+                    mapboxMap.locationComponent.lastKnownLocation!!.longitude
+                ), 5.0
+            )
+        )
+    }
+
+    private fun solarSync() {
+        solarViewModel.fetchSolarsAmerica()
+        solarViewModel.fetchSolarsAsia()
+
+        solarViewModel.solarLiveData.observe(this, androidx.lifecycle.Observer { solarList ->
```

Comment:
observe powinien być w oncreate ponieważ jak klikacie w button to za każdym razem dodaje kolejny obserwator do livedata. Albo czyście obserwatory przed dodaniem nowego albo przenieście. Same fetche opakowałbym w jedną metodą (wzorzec fasada) i tylko raz aktualizował wtedy livedata.